rdtools/normalization.py

```diff
@@ -1,63 +1,125 @@
-''' Energy Normalization Module
-
-This module contains functions to help normalize AC energy output with measured
-poa_global in preparation for calculating PV system degradation.
-'''
+'''Functions for normalizing, rescaling, and regularizing PV system data.'''
 
 import pandas as pd
 import pvlib
 import numpy as np
 from scipy.optimize import minimize
 import warnings
-
+from rdtools._deprecation import deprecated
 
 class ConvergenceError(Exception):
+    '''Rescale optimization did not converge'''
     pass
 
+def normalize_with_expected_power(pv, power_expected, poa_global,
+                                  pv_input='power'):
+    '''
+    Normalize pv output based on expected PV power.
```

Comment:
I think that #217 should help with this to some extent, but I see that there is still room for improvement in the docs here. I will take a swing at that.

Regarding your suggestion to get rid of `pv_input`. That is an interesting suggestion and would simplify things for users. Thinking this through, we want to support the following use cases:

1. Power and expected power on the same or similar time scale (I think this is very common)
2. Energy and expected energy on the same or similar time scale.
3. Energy at low time resolution (e.g. weekly) but expected power at high time resolution (e.g 30 minute from a model using satellite data)

Perhaps we should make some wrappers for cases 1 and 2 (and maybe 3?) that use the present function in the correct way. We'd have to make some modifications to allow `power_expected` to be an energy though.

The insolation is needed for some subsequent calculations (especially soiling), and there is some value in it going through the exact same aggregation as `power_expected`. I think it would be good to make `poa_global` optional in the future (and not provide insolation in that case).

Reply:
I've opened an issue about this so we keep it on the radar.